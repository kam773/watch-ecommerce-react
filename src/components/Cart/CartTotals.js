import React from "react";
import { Link } from "react-router-dom";
import PayPalButton from "./PayPalButton";
import { Container, Row, Col, Button } from "reactstrap";

export default function CartTotals({ value }) {
  const { cartSubTotal, cartTax, cartTotal, clearCart } = value;
  return (
    <React.Fragment>
      <Container>
        <Row>
          <Col
            xs="10"
            className="mt-2 ml-sm-5 ml-md-auto col-sm-8 text-capitalize text-right"
          >
            <Link to="/">
              <Button
                outline
                color="danger"
                className="text-uppercase mb-3 px-5"
                type="button"
                onClick={() => clearCart()}
              >
                clear cart
              </Button>
            </Link>
            <h5>
              <span className="text-title">subtotal :</span>
              <strong>$ {cartSubTotal}</strong>
            </h5>
            <h5>
              <span className="text-title">tax :</span>
              <strong>$ {cartTax}</strong>
            </h5>
            <h5>
              <span className="text-title">total :</span>
              <strong>$ {cartTotal}</strong>
            </h5>
            <PayPalButton
              total={cartTotal}
              clearCart={clearCart}
              history={history}
            />
          </Col>
        </Row>
      </Container>
    </React.Fragment>
  );
}
