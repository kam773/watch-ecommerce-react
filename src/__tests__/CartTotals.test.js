import React from 'react';
import CartTotals from '../components/Cart/CartTotals';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { StaticRouter } from 'react-router'

describe("CartTotals", () => {
    it('should render without crash', () => {
        const props = {
            value: {},
            context: {}
        }
        const tree = renderer.create(
            <StaticRouter>
                <CartTotals {...props} />
            </StaticRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();

    })
    it('should contain a home link', () => {
        const props = {
            value: {}
        }
        const wrapper = shallow(<CartTotals {...props}/>);
        expect(wrapper.find(Link).filter({ to: '/' })).toHaveLength(1);
    });
    it('has 3 h5 tags', () => {
        const props = {
            value: {}
        }
        const wrapper = shallow(<CartTotals {...props}/>);
        expect(wrapper.find('h5')).toHaveLength(3);
    })
})