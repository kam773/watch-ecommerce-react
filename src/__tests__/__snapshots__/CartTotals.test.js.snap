// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotals should render without crash 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-10 mt-2 ml-sm-5 ml-md-auto col-sm-8 text-capitalize text-right"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn-outline-danger text-uppercase mb-3 px-5"
          onClick={[Function]}
          type="button"
        >
          clear cart
        </button>
      </a>
      <h5>
        <span
          className="text-title"
        >
          subtotal :
        </span>
        <strong>
          $ 
        </strong>
      </h5>
      <h5>
        <span
          className="text-title"
        >
          tax :
        </span>
        <strong>
          $ 
        </strong>
      </h5>
      <h5>
        <span
          className="text-title"
        >
          total :
        </span>
        <strong>
          $ 
        </strong>
      </h5>
      <div>
        
      </div>
    </div>
  </div>
</div>
`;
